### PR TITLE
Remove Security Group Rules when deletion failed with DependencyViolation exception

### PR DESCRIPTION
**Pull request by abikouo**
Update terminator Class Ec2SecurityGroup

### CODE REVIEW

File: aws/terminator/networking.py [hunk 1]
```diff
@@ -414,8 +414,23 @@ def name(self):
...
+        )
+
+        # Revoke Ingress rules
+        self.client.revoke_security_group_ingress(
+            GroupId=self.id, IpPermissions=self.instance.get("IpPermissions")
+        )
+
     def terminate(self):
-        self.client.delete_security_group(GroupId=self.id)
+        try:
```
Comment (goneri) [inline aws/terminator/networking.py:429]: What happen if we just call the two methods without the if condition? This way we avoid the risk of breakage if the format of the exception change in the future. I imagine we may need to wrap `revoke_sg_rules()` in `try/except` block thus.
```
self.revoke_sg_rules()
self.client.delete_security_group(GroupId=self.id)
```

**Review by goneri — APPROVED** [review]

File: aws/terminator/networking.py (hunk 1 above)
Comment (abikouo) [reply aws/terminator/networking.py:429]: This was my first idea, but we want to cut down on unnecessary requests

**Comment by jillr** [comment]
The code looks fine but I'm having trouble reproducing the original failure with the eks_cluster tests.  (I'm commenting out the SG and rule cleanup tasks at the end of full_test.yml then letting the terminator handle the SGs).  I'll test more tomorrow, the tests are a bit slow and it's getting late.  :)

**Review by jillr — APPROVED** [review]
This time I was able to reproduce the failure and confirm that this branch solves the issue.  LGTM, deploying.

**Pull request merged**